### PR TITLE
feat(base-image): Move conda to llb and cache it

### PR DESCRIPTION
**Pull request by gaocegege**
Ref #667 

This PR is to:

- Move conda installation into ir package (which means that it is re-implemented with llb.State)
- Add default remote cache for python 3.9 in base-images/remote-cache.
- Add envd version tag into base images.
- Support building remote cache and base images in CI

### CODE REVIEW

**Review by aseaday — COMMENTED** [review]
LGTM

**Review by aseaday — APPROVED** [review]

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -64,16 +64,9 @@ jobs:
       matrix:
         os: [ubuntu-latest]
     runs-on: ${{ matrix.os }}
-    env:
```
[the remark this answers is not in the document]
Comment (gaocegege) [reply .github/workflows/CI.yml:67]: It is to make the action actions-ecosystem/action-get-latest-tag@v1 happy.

**Review by VoVAllen — APPROVED** [review]
LGTM

File: base-images/build.sh [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+set -e
```
[the remark this answers is not in the document]
Comment (gaocegege) [reply base-images/build.sh:3]: LGTM

**Review by kemingy — APPROVED** [review]

**Pull request merged**